vassal-app/src/main/java/VASSAL/build/module/map/Flare.java

```diff
@@ -0,0 +1,346 @@
+  }
+
+  public String[] getAttributeDescriptions() {
+    return new String[] { Resources.getString("Editor.Flare.flare_key"), 
+                          Resources.getString("Editor.Flare.circle_size"), 
+                          Resources.getString("Editor.Flare.circle_color"),
+                          Resources.getString("Editor.Flare.pulses"),
+                          Resources.getString("Editor.Flare.pulses_per_sec") };
+  }
+
+  public String getAttributeValueString(final String key) {
+    if (FLARE_KEY.equals(key)) {
+      return flareKey;
+    }
+    else if (CIRCLE_SIZE.equals(key)) {
+      return String.valueOf(this.circleSize);
+    }
+    else if (CIRCLE_COLOR.equals(key)) {
+      return ColorConfigurer.colorToString(this.color);
+    }
+    else if (PULSES.equals(key)) {
+      return String.valueOf(this.pulses);
+    }
+    else if (PULSES_PER_SEC.equals(key)) {
+      return String.valueOf(this.pulsesPerSec);
+    }
+    return null;
+  }
+  
+  
+  public void setAttribute(String key, Object value) {
+    if (FLARE_KEY.equals(key)) {
+      if (FLARE_ALT_LOCAL.equals(value)) {
+        flareKey = FLARE_ALT;
+      } 
+      else if (FLARE_COMMAND_LOCAL.equals(value) || FLARE_CTRL_LOCAL.equals(value)) {
+        flareKey = FLARE_CTRL;
+      } 
+      else {
+        flareKey = (String) value;
+      }
+    }
+    else if (CIRCLE_SIZE.equals(key)) {
+      if (value instanceof String) {
+        circleSize = Integer.parseInt((String) value);
+      } 
+      else if (value instanceof Integer) {
+        circleSize = (Integer) value;
+      }
+    } 
+    else if (CIRCLE_COLOR.equals(key)) {
+      if (value instanceof String) {
+        value = ColorConfigurer.stringToColor((String) value);
+      }
+      color = (Color)value;
+    }
+    else if (PULSES.equals(key)) {
+      if (value instanceof String) {
+        pulses = Integer.parseInt((String) value);
+      } 
+      else {
+        pulses = (Integer) value;
+      }
+    }
+    else if (PULSES_PER_SEC.equals(key)) {
+      if (value instanceof String) {
+        pulsesPerSec = Integer.parseInt((String) value);
+      } 
+      else {
+        pulsesPerSec = (Integer) value;
+      }
+    }
+  }
+  
+  
+  @Override
+  public HelpFile getHelpFile() {
+    return HelpFile.getReferenceManualPage("Flare.htm");  //$NON-NLS-1$
+  }
+  
+
+  public void addTo(final Buildable parent) {
+    if (parent instanceof Map) {
+      map = (Map) parent;
+      GameModule.getGameModule().addCommandEncoder(this);
+      map.addDrawComponent(this);
+      map.addLocalMouseListener(this);
+      timer  = new Timer(CIRCLE_RATE, this);
+      frame  = 0;
+    }
+  }
+  
+  public void removeFrom(final Buildable parent) {
+    if (parent instanceof Map) {
+      GameModule.getGameModule().removeCommandEncoder(this);
+    }
+  }
+
+
+  public void startAnimation(final boolean isLocal) {
+    if (!isLocal) {
+      map.centerAt(this.clickPoint);
+    }
+    active = true;
```

Comment:
Should the centring respect the 'Centre on Opponents Moves' preference? If a player has turn off move centering for whatever reason, this could potentially be very annoying. My feeling is it should respect the preference. You can use 

if (GlobalOptions.getInstance().centerOnOpponentsMove()) {

What do think about adding a Message Format for reporting Flares? 

<Bob(Germany)> dropped a map flare at U01.03

This would allow the person who has map re-centering turned off to at least know they happened.

Reply:
At the risk of adding features no-one has asked for, it would be neat to have an option so that a text input box pops up when you request a flare and you can type text in and this text appears in the message format, and hovers over the Flare. Of course, players can just type that text into the chat box. Wait for someone to ask for it :)